Framework/Algorithms/src/Q1DWeighted.cpp

```diff
@@ -154,199 +139,228 @@ void Q1DWeighted::exec() {
     wedgeFullAngle *= 2;
   }
 
-  // Create wedge workspaces
-  std::vector<MatrixWorkspace_sptr> wedgeWorkspaces;
-  for (int iWedge = 0; iWedge < nWedges; iWedge++) {
-    double centerAngle = wedgeFullAngle / nWedges * iWedge;
-    centerAngle += wedgeOffset;
-
-    MatrixWorkspace_sptr wedgeWs =
-        WorkspaceFactory::Instance().create(inputWS, 1, sizeOut, sizeOut - 1);
-    wedgeWs->getAxis(0)->unit() =
-        UnitFactory::Instance().create("MomentumTransfer");
-    wedgeWs->setYUnitLabel("1/cm");
-    wedgeWs->setDistribution(true);
-    wedgeWs->setBinEdges(0, XOut);
-    wedgeWs->mutableRun().addProperty("wedge_angle", centerAngle, "degrees",
-                                      true);
-    wedgeWorkspaces.push_back(wedgeWs);
-  }
-
-  // Count histogram for wedge normalization
-  std::vector<std::vector<double>> wedgeXNormLambda(
-      nWedges, std::vector<double>(sizeOut - 1, 0.0));
+  // get the number of wavelength bins in the input, note that the input is a
+  // histogram
+  const size_t nLambda = inputWS->readX(0).size() - 1;
+
+  // we store everything in 3D arrays
+  // index 1 : is for the wedges + the one for the full integration,
+  //           if there are no wedges, the 1st dimension will be 1
+  // index 2 : will iterate over lambda bins
+  // index 3 : will iterate over Q bins
+  // we want to do this, since we want to average the I(Q) in each lambda bin
+  // then average all the I(Q)s together
+  std::vector<std::vector<std::vector<double>>> intensities(
+      nWedges + 1,
+      std::vector<std::vector<double>>(nLambda, std::vector<double>(nQ, 0.0)));
+
+  // the same for the errors
+  std::vector<std::vector<std::vector<double>>> errors(
+      nWedges + 1,
+      std::vector<std::vector<double>>(nLambda, std::vector<double>(nQ, 0.0)));
+
+  // the same dimensions for the normalisation weights
+  std::vector<std::vector<std::vector<double>>> normalisation(
+      nWedges + 1,
+      std::vector<std::vector<double>>(nLambda, std::vector<double>(nQ, 0.0)));
```

Comment:
Can you just copy `errors` and `normalisation` from `intensities`, to avoid the repeated lengthy init code?